### PR TITLE
Enables UTs for AspNetCore.

### PR DESCRIPTION
**Pull request by robward-ms**
### Issues

This pull request makes progress on issues #975, #939, #772, #628, #229.

### Description
This change enables a large set of UTs for AspNetCore. The current pass rate is:

AspNet    : 8230 pass, 0 fail - 100% enabled, 100% pass
AspNetCore: 7057 pass, 224 fail - 88% enabled, 97% pass

The UTs for AspNetCore compile the same test code that is used in
AspNet. To facilitate this, abstraction are used where possible to abstract
away the platform details from the test code. For instance, to create a request,
the UTs use a request factory:

`    var request = RequestFactory.Create(HttpMethod.Get, "http://localhost");`

This will create either an HttpRequest or an HttpRequestMessage depnding on which
platform is being compiled. In this case, the "var" keyword is used since it may be
either or two times at compile time.

For abstractions that are used in multiple tests, a factory has been created in the
Microsoft.Test.AspNet.OData.Factories namespace, the Common\Factories folder. For
an abstraction used in a single test, it's written in the test class itself. These
abstraction make use of conditional compilation using the `NETCORE `and `NETFX `symbols.

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### Additional work necessary

Port E2E Tests to NetCore

### CODE REVIEW

**Comment by robward-ms** [comment]
Failing the CI check is expected; I changed some solution names. This change will required build definition changes.

File: test/UnitTest/Microsoft.Test.AspNet.OData/Common/Types/ISerializableType.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+
+namespace Microsoft.Test.AspNet.OData.Common.Types
+{
```
Comment (AlanWong-MS) [inline test/UnitTest/Microsoft.Test.AspNet.OData/Common/Types/ISerializableType.cs]: Should leave a comment regarding when to un-comment.

File: test/UnitTest/Microsoft.Test.AspNet.OData/Extensions/HttpContentExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            try
+            {
+                JObject obj = JsonConvert.DeserializeObject<JObject>(json);
+                return obj["value"].ToString();
+            }
+            catch (JsonReaderException)
+            {
+            }
+
+            return json;
```
Comment (AlanWong-MS) [inline test/UnitTest/Microsoft.Test.AspNet.OData/Extensions/HttpContentExtensions.cs]: Leftover code from testing?

File: test/UnitTest/Microsoft.Test.AspNet.OData/Routing/ODataPathRouteConstraintTest.cs [hunk 3]
```diff
@@ -120,50 +141,42 @@ public void Match_ReturnsFalse_IfODataPathRouteValueMissing()
...
-            request.SetConfiguration(configuration);
-
+            var routeRequest = new TestRouteRequest(HttpMethod.Get, "http://any/NotAnODataPath");
             var values = new Dictionary<string, object>() { { "odataPath", "NotAnODataPath" } };
             var constraint = CreatePathRouteConstraint();
 
             // Act & Assert
-            Assert.False(constraint.Match(request, null, null, values, HttpRouteDirection.UriResolution));
+            Assert.False(ConstraintMatch(constraint, routeRequest, values, RouteDirection.UriResolution));
+
```
Comment (AlanWong-MS) [inline test/UnitTest/Microsoft.Test.AspNet.OData/Routing/ODataPathRouteConstraintTest.cs]: nit: empty line

File: test/UnitTest/Microsoft.Test.AspNet.OData/Routing/ODataPathRouteConstraintTest.cs [hunk 4]
```diff
@@ -194,7 +203,8 @@ public void Match_DeterminesExpectedServiceRoot_ForMetadata(string prefixString)
             };
 
             // Act
-            var matched = constraint.Match(request, null, null, values, HttpRouteDirection.UriResolution);
+            var matched = ConstraintMatch(constraint, routeRequest, values, RouteDirection.UriResolution);
+
```
Comment (AlanWong-MS) [inline test/UnitTest/Microsoft.Test.AspNet.OData/Routing/ODataPathRouteConstraintTest.cs]: nit: extra empty line

**Pull request merged**